### PR TITLE
incorporate-score-query

### PR DESCRIPTION
**Pull request by josh8551021**
Add metricsScore field to the RestaurantHeader class.
Alters the search queries sent to Elasticsearch to use boosting based on the RestaurantHeader metricsScore field.

### CODE REVIEW

File: capstone/src/main/java/com/google/step/data/RestaurantHeader.java [hunk 1]
```diff
@@ -29,12 +31,18 @@
   private final String name;
   private final GeoPt location;
   private final List<String> cuisine;
+  private final double metricsScore;
 
-  public RestaurantHeader(long restaurantKey, String name, GeoPt location, List<String> cuisine) {
+  public RestaurantHeader(long restaurantKey, String name, GeoPt location, List<String> cuisine, Optional<Double> metricsScore) {
```
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/data/RestaurantHeader.java:36]: What's the general purpose of including this here? Presumably, scores should only travel in one direction (from your calculation to the search indexes), and from there, it should only ever be used in search queries to help sort by relevance.

Is this mainly to address the "isStruggling" stuff? (in which case, this is fine).

File: capstone/src/main/java/com/google/step/search/ElasticsearchClient.java [hunk 2]
```diff
@@ -78,37 +78,40 @@ public void updateRestaurantHeader(RestaurantHeader restaurantHeader) throws IOE
...
     List<String> requestPath = Arrays.asList("", RESTAURANTS, "_search");
 
-    String requestBody = new JSONObject()
-        .put("query", new JSONObject()
-            .put("multi_match", new JSONObject()
-                .put("query", query)
-                .put("fields", new JSONArray(Arrays.asList("name", "cuisine")))))
-        .toString();
+    JSONObject queryRequest = query.isEmpty() ? createMatchAllQuery() : createBasicSearchQuery(query);
+    String requestBody = addBoostingToQuery(queryRequest).toString();
```
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/search/ElasticsearchClient.java:84]: Nice work fixing this logic, this is exactly the kind of thing I was hoping for.

File: capstone/src/main/java/com/google/step/data/RestaurantHeader.java (hunk 1 above)
Comment (josh8551021) [reply capstone/src/main/java/com/google/step/data/RestaurantHeader.java:36]: Yeah, that was the idea. I was under the impression that the score was going to have front-end implications the whole time, we just hadn't had any scores before.

Comment (tzsebe) [reply capstone/src/main/java/com/google/step/data/RestaurantHeader.java:36]: Ok that's fine. Chat with Josh about it, we had discussed a translation into something more meaningful for the frontend, but maybe that translation happens outside of the headers, which is fine too.

Comment (josh8551021) [reply capstone/src/main/java/com/google/step/data/RestaurantHeader.java:36]: Did you mean to mention Martin here?

**Pull request merged**